remoulade/results/backends/redis.py

```diff
@@ -78,36 +95,55 @@ def get_result(self, message_id: str, *, block=False, timeout=None, forget=False
 
         message_key = self.build_message_key(message_id)
         timeout = int(timeout / 1000)
-        if block and timeout > 0:
-            if forget:
-                data = self.client.brpoplpush(message_key, message_key, timeout=timeout)
-                if data:
-                    with self.client.pipeline() as pipe:
-                        pipe.lpushx(message_key, self.encoder.encode(ForgottenResult.asdict()))
-                        pipe.ltrim(message_key, 0, 0)
-                        pipe.execute()
+        deadline = time.monotonic() + timeout
+        retry_count, data = 0, None
+        while data is None:
+            try:
+                if block and timeout > 0:
+                    if forget:
+                        data = self.client.brpoplpush(message_key, message_key, timeout=timeout)
+                        if data:
+                            with self.client.pipeline() as pipe:
+                                pipe.lpushx(message_key, self.encoder.encode(ForgottenResult.asdict()))
+                                pipe.ltrim(message_key, 0, 0)
+                                pipe.execute()
+                        else:
+                            data = None
+                    else:
+                        data = self.client.brpoplpush(message_key, message_key, timeout=timeout)
+
                 else:
-                    data = None
-            else:
-                data = self.client.brpoplpush(message_key, message_key, timeout)
-
-            if data is None:
-                raise ResultTimeout(message_id)
-
-        else:
-            if forget:
-                with self.client.pipeline() as pipe:
-                    pipe.rpushx(message_key, self.encoder.encode(ForgottenResult.asdict()))
-                    pipe.lpop(message_key)
-                    data = pipe.execute()[1]
-            else:
-                data = self.client.rpoplpush(message_key, message_key)
-
-        if data is None:
-            if block:
-                raise ResultTimeout(message_id)
-            else:
-                raise ResultMissing(message_id)
+                    if forget:
+                        with self.client.pipeline() as pipe:
+                            pipe.rpushx(message_key, self.encoder.encode(ForgottenResult.asdict()))
+                            pipe.lpop(message_key)
+                            data = pipe.execute()[1]
+                    else:
+                        data = self.client.rpoplpush(message_key, message_key)
+
+                if data is None:
+                    if block:
+                        raise ResultTimeout(message_id)
+                    else:
+                        raise ResultMissing(message_id)
+
+            except (redis.ConnectionError, redis.TimeoutError):
+                # if data is not None, it means the second step of block+forget ha failed, we can live without a forget
```

Comment:
if data is not None, it means the second step of block+forget **has** failed, we can live without a forget